lib/koans/09_pattern_matching.ex

```diff
@@ -0,0 +1,62 @@
+defmodule PatternMatching do
+  use Koans
+
+  koan "one matches one" do
+    assert match?(1, 1)
+  end
+
+  koan "patterns can be used to pull things apart" do
+    [head | _tail] = [1,2,3,4]
+
+    assert head == 1
+  end
+
+  koan "or put them back together" do
+    head = 1
+    tail = [2,3,4]
+
+    assert [1,2,3,4] == [head | tail]
+  end
+
+  koan "strings come apart just a easily" do
+    "Shopping list: " <> items = "Shopping list: eggs, milk"
+
+    assert items == "eggs, milk"
+  end
+
+  koan "patterns show what you really care about" do
+    %{make: make} = %{type: "car", year: 2016, make: "Honda", color: "black"}
+
+    assert make == "Honda"
+  end
+
+  koan "the pattern can make assertions about what it expects" do
+    the_list = [1, 2, 3]
+
+    assert match?([1, _, _], the_list)
+    refute match?([2, _, _], the_list)
+  end
+
+  def make_noise(%{type: "cat"}), do: "Meow"
+  def make_noise(%{type: "dog"}), do: "Woof"
+  def make_noise(_), do: "Eh?"
+
+  koan "functions can declare what kind of arguments they accept" do
+    dog = %{type: "dog", legs: 4, age: 9, color: "brown"}
+    cat = %{type: "cat", legs: 4, age: 3, color: "grey"}
+    snake = %{type: "snake", legs: 0, age: 20, color: "black"}
+
+    assert make_noise(dog) == "Woof"
+    assert make_noise(cat) == "Meow"
+    assert make_noise(snake) == "Eh?"
+  end
+
+  koan "errors are shaped differently than sucessful results" do
+    result = case Integer.parse("obviously not a number") do
+      {parsed, _} -> parsed
```

Comment:
`Integer.parse` is actually weird
<img width="542" alt="screen shot 2016-03-07 at 11 02 06" src="https://cloud.githubusercontent.com/assets/3731516/13597587/15d20d6c-e522-11e5-8193-6c9781cf6f03.png">


Maybe it's not the best function to show off error handling because you're right, `{:ok, result}` is way more common. Any ideas for what standard library function we could use here to demonstrate patterns in error handling?